### PR TITLE
feat: recommend node compat on builtin use

### LINKED ISSUE #989: feat: trying to use node builtins should log a message recommending to try `node_compat = true`
When you try to use a node builtin right now, esbuild logs an error about setting `platform` to `node`. The message should instead recommend setting `node_compat = true` in their wrangler.toml

### PR DESCRIPTION
**Pull request by rozenmd**
TODO:
- [x] Figure out if the example node app is in the right place
- [x] Write tests!

Closes #989

### CODE REVIEW

File: packages/wrangler/src/bundle.ts
Comment (rozenmd) [inline packages/wrangler/src/bundle.ts:64]: Do we have a better place to put esbuild plugins?

Comment (threepointone) [inline packages/wrangler/src/bundle.ts:64]: hoist this to outside bundleWorker()
```suggestion
 // Add some whitespace and description of what this plugin is 
 /**
 * This plugin's description
 */
  const checkForNodeBuiltinsPlugin = {
```

Comment (threepointone) [reply packages/wrangler/src/bundle.ts:64]: It's fine inside here for now

**Review by threepointone — COMMENTED** [review]
This needs a test! Write a test that tries to bundle a worker with a node built in, and make sure it throws the error you specified

**Comment by threepointone** [comment]
The windows CI issue is because of https://github.com/cloudflare/wrangler2/pull/977, investigating that separately

File: packages/wrangler/src/bundle.ts
Comment (rozenmd) [reply packages/wrangler/src/bundle.ts:73]: oops 🤦

**Review by threepointone — APPROVED** [review]
Looks good to me! This should be a patch changeset btw. Approving for now, but please don't land until we fix the windows CI issue, thanks!

File: packages/wrangler/src/bundle.ts
Comment (JacobMGEvans) [inline packages/wrangler/src/bundle.ts:39]: Should use the logger Error utility

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]

File: packages/wrangler/src/bundle.ts
Comment (threepointone) [reply packages/wrangler/src/bundle.ts:39]: We went it to throw an error though, not just log

Comment (JacobMGEvans) [reply packages/wrangler/src/bundle.ts:39]: `throw` the `logger.error` or we should add the ability to pass a option that has the logger `throw Error` seems like it should somehow be consistent and utilize the utility, I will unblock the PR either way.

**Review by JacobMGEvans — APPROVED** [review]

File: packages/wrangler/src/bundle.ts
Comment (threepointone) [reply packages/wrangler/src/bundle.ts:39]: I don't understand, the logger only logs, but we want this to be an exception. We log the error to the terminal the error when it gets caught anyway. Where are we missing consistency?

Comment (JacobMGEvans) [reply packages/wrangler/src/bundle.ts:39]: > I don't understand, the logger only logs, but we want this to be an exception. We log the error to the terminal the error when it gets caught anyway. Where are we missing consistency?

`logger.error` seems redundant then, you can also throw anything. Consistency is lost in how we console out log errors vs exceptions errors, one has a custom format the other is default system format. Not causing a SIGTERM for some console errors makes sense I suppose maybe certain validation so you can collect all the errors before SIGTERM, I just don't understand why they need to look different too?

Comment (JacobMGEvans) [reply packages/wrangler/src/bundle.ts:39]: Wait maybe I missed this, 
> We log the error to the terminal the error when it gets caught anyway.

All Errors are now passed into the `logger.error`?

Comment (threepointone) [reply packages/wrangler/src/bundle.ts:39]: We do! https://github.com/cloudflare/wrangler2/blob/a6e20571e3f7ca2b5a21a0140c9a938120447b2d/packages/wrangler/src/index.tsx#L2653-L2671

Comment (threepointone) [reply packages/wrangler/src/bundle.ts:39]: @rozenmd could you add a small screenshot of what the error looks like?

Comment (rozenmd) [reply packages/wrangler/src/bundle.ts:39]: @threepointone: 
image

**Pull request merged**
